internal/config/config.go

```diff
@@ -29,7 +29,7 @@ type Config interface {
 }
 
 func NewConfig() (Config, error) {
-	c, err := ghConfig.Read()
+	c, err := ghConfig.Read(fallbackConfig())
```

Comment:
As this is using a `sync.Once` underneath the covers, I could not think of a great way to test this. I am open to ideas here. I manually tested it and everything worked as expected.

Reply:
Something like:

```
func TestNewConfigProvidesFallback(t *testing.T) {
    var spiedConfig *ghConfig.Config
    ghConfig.Read = func (fallback *ghConfig.Config) (*ghConfig.Config, error) {
        spiedConfig = fallback
        return nil, nil
    }
    
    requireKeyWithValue(t, spiedCfg, []string{"git_protocol"}, "https")
}
```

I just wrote this in the comment box so it's probably janky